### ISSUE TITLE
Restructure ToC

### ISSUE THREAD

**Issue by pcraig3** [HHS/simpler-grants-pdf-builder#105]
> The TOC needs a little restructuring, but all the components are there which is fantastic!

> I noticed all the links have a <Reference> and the a nested <Link>, you don’t need both since it is just repetitive – you can just have <Link>

| before | after  |
|--------|--------|
|   TOC > TOCI > TOC > TOCI > Reference > Link > Link     |  TOC > TOCI > Reference > Link      |
| <img width="658" alt="Image" src="https://github.com/user-attachments/assets/dc3e9d35-3d0a-4599-879f-15a203d136d7" /> | <img width="658" alt="Image" src="https://github.com/user-attachments/assets/b612f2d7-5f70-4494-ad69-a58804f90fba" /> |

**Comment by pcraig3** [comment]
Update to this issue now that we have talked it through. 

Basically, I am not sure if this means we should create a flat hierarchy in our table of contents, or if the Sections themselves should be grouped along with their Subsection headings as a sublist.

Here is an illustration of the two alternatives.

| Option 1 | Option 2  |
|--------|--------|
| flat list of ToC links       |   "Step" headings are used to group links below them     |
| ![Image](https://github.com/user-attachments/assets/09bcc171-0300-4b72-8ea6-359a561ae025) | ![Image](https://github.com/user-attachments/assets/9530b90f-b33e-4a2b-8a4e-baaf5eb56647) |

**Comment by pcraig3** [comment]
Got a response back from Kate that said:

> Option 2, you want to use headings to group sections and sub-sections

So we went with option 2!

PR just in #183, so closing this.
